make/test.js

```diff
@@ -0,0 +1,33 @@
+const assert = require('assert');
+const make = require('./make.js');
+
+function sum(a, b) {
+  return a + b;
+}
+
+function multiply(a, b) {
+  return a * b;
+}
+
+describe('make', () => {
+  it('1', () => {
+    const check = make(15)(34, 21, 666)(41)(sum);
+    assert.deepEqual(check, 777);
+  });
+  it('2', () => {

```

Comment:
All your `it` blocks tests similar behavior and should be combined in one `it`.
Also you should add some edge case testing and failure testing if rules of function isn't followed.